browser/static/browser/js/elastic_browser.js

```diff
@@ -309,21 +314,32 @@ var ElasticBrowser = (function () {
                             })
                     }
 
-                    if (desc !== "HIDE DIRECTORY") {
-                        dir_results_string = dir_results_string + Mustache.render(
-                            dir_template,
-                            {
-                                path: dir_array[i]._source.path,
-                                item: dir_array[i]._source.dir,
-                                description: desc,
-                                size: "",
-                                actions: ""
-                            }
-                        )
+                    if (desc !== "HIDE DIRECTORY" && dir_array[i]._source.url !== undefined) {
+                            dir_results_string = dir_results_string + Mustache.render(
+                                dir_template,
+                                {
+                                    path: dir_array[i]._source.path,
+                                    item: dir_array[i]._source.dir,
+                                    description: desc,
+                                    size: "",
+                                    actions: info_templ
+                                }
+                            )
+                        }
+                    else if (desc !== "HIDE DIRECTORY" && dir_array[i]._source.url == undefined) {
```

Comment:
I would remove the `&& dir_array[i]._source.url == undefined` as you have covered the other cases in the if above.

Also, just as an aside, in javascript it is a good idea to use  === instead of ==. Here is a little description why:

> To quote Douglas Crockford's excellent JavaScript: The Good Parts,
> 
> JavaScript has two sets of equality operators: === and !==, and their evil twins == and !=. The good ones work the way you would expect. If the two operands are of the same type and have the same value, then === produces true and !== produces false. The evil twins do the right thing when the operands are of the same type, but if they are of different types, they attempt to coerce the values. the rules by which they do that are complicated and unmemorable. These are some of the interesting cases:
> 
> '' == '0'           // false
> 0 == ''             // true
> 0 == '0'            // true
> 
> false == 'false'    // false
> false == '0'        // true
> 
> false == undefined  // false
> false == null       // false
> null == undefined   // true
> 
> ' \t\r\n ' == 0     // true
> The lack of transitivity is alarming. My advice is to never use the evil twins. Instead, always use === and !==. All of the comparisons just shown produce false with the === operator.